ale_linters/cpp/clangtidy.vim

```diff
@@ -0,0 +1,15 @@
+" Author: vdeurzen <tim@kompiler.org>
+" Description: clang-tidy linter for cpp files
+
+" Set this option to change the clang-tidy options for warnings for C.
+if !exists('g:ale_cpp_clangtidy_options')
+    let g:ale_cpp_clangtidy_options = '-std=c++14 -Wall'
+endif
+
+call ale#linter#Define('cpp', {
+\   'name': 'clangtidy',
+\   'output_stream': 'stdout',
+\   'executable': 'clang-tidy',
+\   'command': g:ale#util#stdin_wrapper . ' -- ' . g:ale_cpp_clangtidy_options,
```

Comment:
If you can figure out some way to allow a special argument for both the Bash and Batch scripts to specify that the filename argument should come at the front of the list of arguments instead of at the end, without breaking everything, then go for it.